### ISSUE TITLE
OpenTelemetry - Provide the capability for `VertxTracingAdapter` to not create spans for specified paths

### ISSUE THREAD

**Issue by jamesnetherton** [quarkusio/quarkus#18708]
### Description

For OpenTelemetry, it'd be useful if it were possible to introduce some mechanism that would enable `VertxTracingAdapter` to not create spans for a set of specified paths? (I.e via a BuildItem that other extensions can leverage)

There's a feature of the Camel OpenTelemetry component where you can exclude specific endpoints from tracing. It works fine, except for when Camel creates routes on the Quarkus Vert.x router. So was hoping we could do something to help with that. There's some more background here https://github.com/apache/camel-quarkus/issues/2897.

It'd be nice if such a path filter could work via exact matches, wilcards and regular expressions as defined [here](https://camel.apache.org/camel-quarkus/latest/reference/extensions/opentelemetry.html#quarkus.camel.opentelemetry.exclude-patterns).

**Comment by brunobat** [comment]
@radcortez Will you be working on this or should I leave it unassigned?

**Comment by radcortez** [comment]
I can look into this.

**Comment by geoand** [comment]
@radcortez would you like me to look into this?

**Comment by radcortez** [comment]
Sure.

**Comment by geoand** [comment]
Just a historical note, `VertxTracingAdapter` was removed in 57f6a9d9cb2

**Comment by geoand** [comment]
Is this issue talking about server spans?

**Comment by radcortez** [comment]
Yes, server spans. The `VertxTracingAdapter` is now https://github.com/quarkusio/quarkus/blob/71c6f7d7df8f85a866e019f6ced1ceb08a74bb05/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/intrumentation/vertx/InstrumenterVertxTracer.java

**Comment by geoand** [comment]
The simplest solution here would probably be to allow extensions to add stuff to `DropTargetsSampler`
